### ISSUE TITLE
Clothing sizes and flags need a bit of reworking

### ISSUE THREAD

**Issue by catdach** [CleverRaven/Cataclysm-DDA#57775]
### Is your feature request related to a problem? Please describe.

Right now clothes come in 3 different sizes: undersized, normal, and oversized. These are implemented in Json via the flags `"UNDERSIZE"` and `"OVERSIZE"` respectively, with normal size being the default. There is also the flag `"VARSIZE"`, which allows the addition of either of the size flags in accordance with the current size of the player; or remove them if the player is normal sized.

Mechanically the `"UNDERSIZE"` flag just makes it so that you don't get extra encumbered by the item if you're tiny (otherwise the encumbrance is [multiplied by 2](https://github.com/CleverRaven/Cataclysm-DDA/blob/556afb9bf7cceca4db5a43c0dcac3a10fd2439bc/src/item.cpp#L7465)), and if you're _not_ tiny then it [multiplies the encumbrance by 3](https://github.com/CleverRaven/Cataclysm-DDA/blob/556afb9bf7cceca4db5a43c0dcac3a10fd2439bc/src/item.cpp#L7460). 

The `"OVERSIZE"` flag on the other hand, simply allows the item to be worn, even if a mutation would normally prevent it from being wearable.

this comes with some problems:

1. Beyond being very different form each other functionally, the `"OVERSIZE"` flag really doesn't have much to with size at all. Sometimes this leads to clothing to be wearable with mutations that really shouldn't be affected by the size of the item. For example the tail mutations prevent the wearing of non-soft pants unless those pants are...extra large???? sometimes it kinda makes sense, an extra large mask would be more accommodating for a mutation like a beak or mandibles I guess.

2. In antithesis of mutations that shouldn't care about item sizes, there are also items that shouldn't care about the size of the player (and usually they shouldn't care about any mutations at all). These are items that would be attached by easily adjustable straps, provide negligible coverage, and are usually for minor storage or other effects ie: holsters, headlamps, watches, maybe even belts and rings. This actually has a workaround, which is just to add `"OVERSIZE"` to these items; which comes with the problem of this ugly little thing being on items that _should_ be perfectly normal sized:
![image](https://user-images.githubusercontent.com/93795131/168696833-e1199636-bd34-4b5c-842f-d6f105f86708.png)
and this really doesn't do anything for when tiny. We could also add `"UNDERSIZE"` but then non-tiny players would get a notable larger amount of encumbrance from it.

3. There is also the case of a player having the tiny mutation _and_ some other mutation that restricts armor. For example the mouse mutation line has a mutation called "rodent muzzle", which restricts gear from being worn on the mouth. So which gasmask are they supposed to wear? The XS one doesn't fit and the XL one has a ton of extra encumbrance. I guess we could also add `"OVERSIZE"` to the XS gasmask, but that's kinda ridiculous. 

4. Despite what is implied by: 
![image](https://user-images.githubusercontent.com/93795131/168696833-e1199636-bd34-4b5c-842f-d6f105f86708.png),
oversized stuff doesn't seem to have any more encumbrance than normal sized stuff. In fact, it probably can't; otherwise the workaround of adding `"OVERSIZE"` to items as a pseudo `"ALLOWS_ALL_MUTATIONS"` flag wouldn't be happening. This is really misleading and confusing to have things that "don't fit" that doesn't have any noticeable mechanical impact.

### Solution you would like.

1. Rename the current `"OVERSIZE"` flag on all non-XL-items to something like `"ALLOWS_ALL_MUTATIONS"`.

This should solve problems 1 and 3.
We already have `"ALLOWS_TAIL"` and `"ALLOWS_TALONS"` flags. Currently `"OVERSIZE"` serves as the catch-all for allowing mutations, which is fine, but it would be nice to separate sizes from the other restrictive mutations. Really it was always a bit of caveman logic to say things like "Mutation make clothes hard to wear? **Make clothes bigger!**"

This also means that the way "huge" mutations work is going to need to change. Being too large are supposed to make small/normal size clothes unwearable. Huge mutations do this by restricting clothes in same way that horn/muzzle mutations restrict headgear. Huge mutations just do that for **every slot.** I would honestly like this to be redesigned into something new, but that should maybe wait until the new limb system is done first. For now it could work like tail mutations by adding a line like this to the huge mutation: `"allowed_items": [ "OVERSIZE", "SIZELESS" ]` (note the intentional lack of `"ALLOWS_ALL_MUTATIONS"`)

2. Add a new flag `"SIZELESS"`.

As stated above in problem 2, some items shouldn't care about the size of the player. Adding this, along with `"ALLOWS_ALL_MUTATIONS"`, should fully solve problem 2. 

**Important:** `"SIZELESS"` and `"ALLOWS_ALL_MUTATIONS"` **should not be combined into one flag!** The whole point is to separate size restrictions as distinct from miscellaneous mutation restrictions. Otherwise we create problem 1 again. Sure, most items that have one will probably have the other, but that isn't _necessarily_ true for _**all**_ items.

3. Make `"OVERSIZE"` work more like `"UNDERSIZE"`

Basically make it extra encumbering for a normal size person to wear oversized gear in the same way that `"UNDERSIZE"` does (Probably to a lesser degree though. Maybe 2 times the encumbrance rather than 3?). 

This should solve problem 4.

### Describe alternatives you have considered.

Throwing out `"UNDERSIZE"` and keeping `"OVERSIZE"` the way it is. This would get rid of problem 3 but still leaves problems 1, 2 and 4.

Waiting for the new limbs and injury systems overhauls as they _could_ be relevant to this stuff somehow?

Waiting until after G. This stuff probably isn't a high priority, so yeah it probably should wait until after G.

### Additional context

Perhaps tangentially related/relevant in some way #54939, #51294, #54872, #56326 ?

**Comment by fairyarmadillo** [comment]
`Throwing out "UNDERSIZE" and keeping "OVERSIZE" the way it is. This would get rid of problem 3 but still leaves problems 1, 2 and 4.`

That's how it used to work. Mice were just SOL (by design) and couldn't wear very much and XL was less of a catch-all because there were fewer recipes and they tended to be worse on encumbrance.

`Basically make it extra encumbering for a normal size person to wear oversized gear in the same way that "UNDERSIZE" does (Probably to a lesser degree though. Maybe 2 times the encumbrance rather than 3?).`

The rigidity system could help out here as well. An XL pair of boots or plate mail should be unusable by a normal human because they're sized for a 10 foot tall hippopotamus man, while you could maybe put on his duster if you didn't mind the skirt dragging behind you.

I've never felt like XL gear was really done with any sort of consistent intentionality. Sometimes it's just the same piece of gear with more weight, sometimes it has different stats, more encumbrance, etc. - it always felt arbitrary.

**Comment by catdach** [comment]
> > Throwing out "UNDERSIZE" and keeping "OVERSIZE" the way it is. This would get rid of problem 3 but still leaves problems 1, 2 and 4.
> 
> That's how it used to work. Mice were just SOL (by design) and couldn't wear very much

Well...kinda? When the mouse mutation line [was introduced](https://github.com/CleverRaven/Cataclysm-DDA/pull/24672), any item with `"VARSIZE"`(most cloth stuff) could be "downsized" with a tailors kit. Still limiting, but not as bad as it could be.

> XL was less of a catch-all because there were fewer recipes and they tended to be worse on encumbrance.

what I mean by catch-all is that some non-XL items like holsters were given the `"OVERSIZE"` flag in as a substitute for the non-existent `"SIZELESS"` and/or `"ALLOWS_ALL_MUTATIONS"` flags. I'm not talking about the explicit XL items.

> > Basically make it extra encumbering for a normal size person to wear oversized gear in the same way that "UNDERSIZE" does (Probably to a lesser degree though. Maybe 2 times the encumbrance rather than 3?).
> 
> The rigidity system could help out here as well. An XL pair of boots or plate mail should be unusable by a normal human because they're sized for a 10 foot tall hippopotamus man, while you could maybe put on his duster if you didn't mind the skirt dragging behind you.

I suppose so? For the sake of consistency I would prefer the relationship of "normal size player to large gear" to be similar to (if not _exactly_ the same as,) the relationship of "tiny size player to normal gear." 

> I've never felt like XL gear was really done with any sort of consistent intentionality. Sometimes it's just the same piece of gear with more weight, sometimes it has different stats, more encumbrance, etc. - it always felt arbitrary.

*sigh* yeah, I'm hoping XS gear doesn't fall down that same path. hopefully #56326 will allow greater consistency or even allow for the obsoleting of specific XL and XS gear in their entirety.

**Comment by Stadler76** [comment]
Is it intended, that a long scarf fits a beak/muzzle but not a scarf? I'm not talking about the OVERSIZE-flag here. It's more, that I wonder, if we might need a clothing flag, like: `ALLOWS_MUZZLE`?
